### ISSUE TITLE
Make function names more readable/friendly

### ISSUE THREAD

**Issue by cocoa-xu** [cocoa-xu/evision#19]
Function names are not quite readable/friendly as I use lowercases for every function to get it worked ASAP. Now it's time to improve it. 

The reason why I use all lowercase is that some C++ function names start with capital letters, some have abbreviations in the names. Some examples, 

## Approach 1

```cpp
cv::BFMatcher::BFMatcher()
// will be illegal in Elixir if we make it OpenCV.BFMatcher.BFMatcher/0
// also, it starts with an abbreviation, BF => brute-force
// so we should use 
// OpenCV.BFMatcher.bfMatcher/0
```

Therefore, we can have the following rules
- For consecutive capital letters, the last one should be kept as is, and the rest of them should be transformed to lowercase, e.g., `cv::BOWImgDescriptorExtractor::BOWImgDescriptorExtractor()` should be `OpenCV.BOWImgDescriptorExtractor.bowImgDescriptorExtractor/0`
- For any single capital letter, leave it as is unless it is the first one.

```
good cases:
 BFMatcher => bfMatcher
 R => r

acceptable:
  getkNNSamples => getknnSamples

bad cases:
 BOWKMeansTrainer => bowkMeansTrainer
 getFPS => getfpS
 TrackerGOTURN_Params => trackergoturN_Params
```

## Approach 2
What about using underscore? 

Rules:

1. For consecutive capital letters, add an `_` right before the last one and then make them lowercase. 
2. For a single upper case letter, prefix it with `_` (unless there exists an `_` before the transformation) and make it lowercase.
3. If the function name consists of only one upper case letter, just make it lowercase.

```
good cases:
 BFMatcher => bf_matcher
 BOWImgDescriptorExtractor => bow_img_descriptor_extractor
 setPreFilterType => set_pre_filter_type
 R => r
 dnn_Model => dnn_model

acceptable:
 getkNNSamples => getknn_samples

bad cases:
 setROI1 => setro_i1
 getFPS => getfp_s
```

## Approach 3
Just make sure the first letter is lowercase.

```
good cases:
 R => r
 getkNNSamples => getkNNSamples
 getFPS => getFPS
 TrackerGOTURN_Params => trackerGOTURN_Params

bad cases:
 BFMatcher => bFMatcher
 BOWKMeansTrainer => bOWKMeansTrainer
```

**Comment by cocoa-xu** [comment]
## Approach 4
Leave it as is if the first letter is lowercase. Otherwise, prefix it with a `_`.

```
good cases:
 getkNNSamples => getkNNSamples
 getFPS => getFPS

acceptable??:
 R => _R
 TrackerGOTURN_Params => _TrackerGOTURN_Params
 BFMatcher => _BFMatcher
 BOWKMeansTrainer => _BOWKMeansTrainer

bad cases:
```

This approach probably has no bad cases, but it feels like functions with an underscore prefix are meant to be private or ignored ones.

**Comment by cocoa-xu** [comment]
Addressed in #27 

Will be using **smallCamelCase** for almost all OpenCV functions.

The reason why I chose **smallCamelCase** is that it transforms much fewer function names as a large portion of them in OpenCV are starting with a lowercase letter.
